Google.PowerShell/PubSub/GcpsMessage.cs

```diff
@@ -0,0 +1,222 @@
+        /// <para type="description">
+        /// The topic to which the messages will be published.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0)]
+        [ValidateNotNullOrEmpty]
+        public string Topic { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Data of the message that will be published.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.DataAndAttributes)]
+        [ValidateNotNullOrEmpty]
+        public string Data { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Attributes of the message that will be published.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.DataAndAttributes)]
+        [ValidateNotNullOrEmpty]
+        public Hashtable Attributes { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Messages to be published. Use this parameter to publish one or more messages.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.Message)]
+        [ValidateNotNullOrEmpty]
+        public PubsubMessage[] Message { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            Topic = GetProjectPrefixForTopic(Topic, Project);
+            if (ParameterSetName == ParameterSetNames.DataAndAttributes)
+            {
+                PubsubMessage psMessage = NewGcpsMessage.ConstructPubSubMessage(Data, Attributes);
+                Message = new PubsubMessage[] { psMessage };
+            }
+
+            // Encode data in each message to base64.
+            foreach (PubsubMessage psMessage in Message)
+            {
+                if (!string.IsNullOrWhiteSpace(psMessage.Data))
+                {
+                    psMessage.Data = Convert.ToBase64String(Encoding.UTF8.GetBytes(psMessage.Data));
+                }
+            }
+
+            PublishRequest requestBody = new PublishRequest() { Messages = Message };
+            ProjectsResource.TopicsResource.PublishRequest publishRequest = Service.Projects.Topics.Publish(requestBody, Topic);
+            try
+            {
+                PublishResponse response = publishRequest.Execute();
+
+                // The MessageIds is a server-assigned ID of the published message.
+                // They are in the same order as the messages in the request.
+                // We will use them to populate each message in the request with the corresponding message id.
+                IList<string> returnedMessageIds = response.MessageIds;
+                if (returnedMessageIds.Count != requestBody.Messages.Count)
+                {
+                    throw new InvalidOperationException($"Only {returnedMessageIds.Count} out of {requestBody.Messages.Count} is published.");
```

Comment:
I think 'is' should be dropped.  

"Only 2 out of 10 published."   and "Only 1 out of 10 published."  

Otherwise I think you will need to use is for singular and are for multiple